src/lib/structures/expr.mli

```diff
@@ -326,8 +335,13 @@ val print_th_elt : Format.formatter -> th_elt -> unit
 
 val is_pure : t -> bool
 
-val const_term : t -> bool
-(** return true iff the given argument is a term without arguments *)
+val is_model_term : t -> bool
+(** [is_model_term e] checks if the expression [e] is a model terms.
+    A model term can be:
+    - A record definition involving only model terms.
+    - A constructor application involving only model terms,
+    - A literal of a basic type (integer, real, boolean, unit or bitvector),
+    - An application of function of arity [0]. *)
```

Comment:
Is this *any* function of arity 0 or a specific subset (I guess internal & abstract IDs? Or maybe just abstract IDs?)

Reply:
I thought it could be only abstract and internal names, but it seems to be any name. For instance, 
```smt2 
(set-option :produce-models true)
(set-logic ALL)
(declare-sort s 0)
(declare-const x s)
(declare-fun f (Int) s)
(assert (distinct (f 0) x))
(check-sat)
(get-model)
```
Here, the class representative of `x` is `x` itself. As we use `Expr.is_model_term` in `X.to_model_term`, the name `x` will occur in `Expr.is_model_term`.

Reply:
Ah yes, we don't generate abstract identifiers for things that are already identifiers, there would be no point!